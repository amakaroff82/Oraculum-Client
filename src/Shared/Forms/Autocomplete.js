import React from 'react';
import { Autocomplete as RtAutocomplete } from 'react-toolbox/lib/autocomplete';
import { withStyles } from 'material-ui/styles';
import { FormControl, FormHelperText } from 'material-ui/Form';

const styles = theme => ({
  autocompleteWrapper: {
    width: '100%',
    '&>div::after': {
      // this is used for the dropdown arrow.
      borderTopColor: theme.palette.input.helperText,
    },
    '& input': {
      borderBottomColor: theme.palette.input.bottomLine,
      color: theme.palette.input.inputText,
    },
    '& span': {
      color: theme.palette.input.helperText,
    },
    '& label': {
      color: theme.palette.input.labelText,
    },
  },
});

export const Autocomplete = ({
  direction,
  suggestionMatch,
  label,
  source,
  disabled,
  multiple,
  classes,
  helperTextClassName,
  helperText,
  FormHelperTextProps,
  required,
  className,
  input: { value, onBlur, onChange },
  meta: { error },
}) => {
  const hasError = !!error;
  const hasHelperText = !!helperText;
  const separator = hasError && hasHelperText ? ' | ' : '';

  return (
    <FormControl
      className={className}
      disabled={disabled}
      error={!!error}
      fullWidth
      margin="normal"
      required={required}
    >
      <RtAutocomplete
        className={classes.autocompleteWrapper}
        direction={direction}
        suggestionMatch={suggestionMatch}
        label={label}
        source={source}
        disabled={disabled}
        onBlur={() => onBlur()}
        onChange={value => onChange(value)}
        multiple={multiple}
        selectedPosition="below"
        value={value}
      />
      {(hasError || hasHelperText) && (
        <FormHelperText
          className={helperTextClassName}
          {...FormHelperTextProps}
        >
          {hasError && error}
          {separator}
          {hasHelperText && helperText}
        </FormHelperText>
      )}
    </FormControl>
  );
};

export default withStyles(styles)(Autocomplete);
