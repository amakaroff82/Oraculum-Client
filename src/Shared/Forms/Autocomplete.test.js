import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Autocomplete } from './Autocomplete';

const defaultProps = {
  classes: {},
  input: { value: ['foo', 'bar'], onBlur: jest.fn(), onChange: jest.fn() },
  meta: {},
};

const setupShallow = props => {
  props = { ...defaultProps, ...props };
  return shallow(<Autocomplete {...props} />);
};

describe('<Autocomplete /> component', () => {
  it('renders with default props correctly', () => {
    const autocomplete = setupShallow();
    expect(toJson(autocomplete)).toMatchSnapshot();
  });

  it('renders with label correctly', () => {
    const autocomplete = setupShallow({ label: 'foo' });
    expect(toJson(autocomplete)).toMatchSnapshot();
  });

  it('renders with helperText correctly', () => {
    const autocomplete = setupShallow({ helperText: 'bar' });
    expect(toJson(autocomplete)).toMatchSnapshot();
  });

  it('renders error correctly', () => {
    const autocomplete = setupShallow({ meta: { error: 'sad' } });
    expect(toJson(autocomplete)).toMatchSnapshot();
  });

  it('renders error, concat with helperText if present', () => {
    const autocomplete = setupShallow({
      helperText: 'bar',
      meta: { error: 'sad' },
    });
    expect(toJson(autocomplete)).toMatchSnapshot();
  });
});
