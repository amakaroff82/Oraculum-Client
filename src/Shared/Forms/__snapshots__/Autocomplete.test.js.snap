// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Autocomplete /> component renders error correctly 1`] = `
<withStyles(FormControl)
  error={true}
  fullWidth={true}
  margin="normal"
>
  <ThemedAutocomplete
    allowCreate={false}
    className=""
    composeTheme="deeply"
    direction="auto"
    keepFocusOnChange={false}
    mapThemrProps={[Function]}
    multiple={true}
    onBlur={[Function]}
    onChange={[Function]}
    selectedPosition="below"
    showSelectedWhenNotInSource={false}
    showSuggestionsWhenValueIsSet={false}
    source={Object {}}
    suggestionMatch="start"
    value={
      Array [
        "foo",
        "bar",
      ]
    }
  />
  <withStyles(FormHelperText)>
    sad
  </withStyles(FormHelperText)>
</withStyles(FormControl)>
`;

exports[`<Autocomplete /> component renders error, concat with helperText if present 1`] = `
<withStyles(FormControl)
  error={true}
  fullWidth={true}
  margin="normal"
>
  <ThemedAutocomplete
    allowCreate={false}
    className=""
    composeTheme="deeply"
    direction="auto"
    keepFocusOnChange={false}
    mapThemrProps={[Function]}
    multiple={true}
    onBlur={[Function]}
    onChange={[Function]}
    selectedPosition="below"
    showSelectedWhenNotInSource={false}
    showSuggestionsWhenValueIsSet={false}
    source={Object {}}
    suggestionMatch="start"
    value={
      Array [
        "foo",
        "bar",
      ]
    }
  />
  <withStyles(FormHelperText)>
    sad
     | 
    bar
  </withStyles(FormHelperText)>
</withStyles(FormControl)>
`;

exports[`<Autocomplete /> component renders with default props correctly 1`] = `
<withStyles(FormControl)
  error={false}
  fullWidth={true}
  margin="normal"
>
  <ThemedAutocomplete
    allowCreate={false}
    className=""
    composeTheme="deeply"
    direction="auto"
    keepFocusOnChange={false}
    mapThemrProps={[Function]}
    multiple={true}
    onBlur={[Function]}
    onChange={[Function]}
    selectedPosition="below"
    showSelectedWhenNotInSource={false}
    showSuggestionsWhenValueIsSet={false}
    source={Object {}}
    suggestionMatch="start"
    value={
      Array [
        "foo",
        "bar",
      ]
    }
  />
</withStyles(FormControl)>
`;

exports[`<Autocomplete /> component renders with helperText correctly 1`] = `
<withStyles(FormControl)
  error={false}
  fullWidth={true}
  margin="normal"
>
  <ThemedAutocomplete
    allowCreate={false}
    className=""
    composeTheme="deeply"
    direction="auto"
    keepFocusOnChange={false}
    mapThemrProps={[Function]}
    multiple={true}
    onBlur={[Function]}
    onChange={[Function]}
    selectedPosition="below"
    showSelectedWhenNotInSource={false}
    showSuggestionsWhenValueIsSet={false}
    source={Object {}}
    suggestionMatch="start"
    value={
      Array [
        "foo",
        "bar",
      ]
    }
  />
  <withStyles(FormHelperText)>
    bar
  </withStyles(FormHelperText)>
</withStyles(FormControl)>
`;

exports[`<Autocomplete /> component renders with label correctly 1`] = `
<withStyles(FormControl)
  error={false}
  fullWidth={true}
  margin="normal"
>
  <ThemedAutocomplete
    allowCreate={false}
    className=""
    composeTheme="deeply"
    direction="auto"
    keepFocusOnChange={false}
    label="foo"
    mapThemrProps={[Function]}
    multiple={true}
    onBlur={[Function]}
    onChange={[Function]}
    selectedPosition="below"
    showSelectedWhenNotInSource={false}
    showSuggestionsWhenValueIsSet={false}
    source={Object {}}
    suggestionMatch="start"
    value={
      Array [
        "foo",
        "bar",
      ]
    }
  />
</withStyles(FormControl)>
`;
